admin_guide/tcp_ingress_external_ports.adoc

```diff
@@ -0,0 +1,130 @@
++
+[CAUTION]
+====
+You must ensure that the IP address pool you assign terminates at one or more
+nodes in your cluster. You can use the existing
+xref:../admin_guide/high_availability.adoc#configuring-ip-failover[`*oadm ipfailover*`] to ensure that the external IPs are highly available.
+====
+
+For manually-configured external IPs, potential port clashes are handled on a
+first-come, first-served basis. If you request a port, it is only available if
+it has not yet been assigned for that IP address. For example:
+
+.Port clash example for manually-configured external IPs
+====
+Two services have been manually configured with the same external
+IP address of 172.7.7.7.
+
+`MongoDB service A` requests port 27017, and then
+`MongoDB service B` requests the same port; the first request gets the port. 
+====
+
+However, port clashes are not an issue for external IPs assigned by the ingress
+controller, because the controller assigns each service a unique address.
+
+[NOTE]
+====
+Ingress IPs can only be assigned if the cluster is not running in the cloud. In
+cloud environments, LoadBalancer-type services configure cloud-specific load
+balancers.
+====
+
+[[unique-external-ips-ingress-traffic-configure-cluster]]
+== Configuring the Cluster to Use Unique External IPs
+
+In non-cloud clusters, `ingressIPNetworkCIDR` is set by default to `172.46.0.0/16`. You could use the default if your cluster environment is not already using this private range. However, if you want to use a different range, then before you assign an ingress IP you must set xref:../install_config/master_node_configuration.adoc#master-node-config-network-config[`ingressIPNetworkCIDR`]
+in the *_/etc/origin/master-config.yaml_* file, then restart the master service.
+
+[[unique-external-ips-ingress-traffic-configure-service]]
+== Configuring an Ingress IP for a Service
+
+To assign an ingress IP:
+
+. Create a YAML file for a LoadBalancer service that requests a specific IP via the `*loadBalancerIP*` setting:
++
+.Sample LoadBalancer Configuration
+====
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: egress-1
+spec: ports:
+  - name: db
+    port: 3306
+  loadBalancerIP: 172.46.0.1
+  type: LoadBalancer
+  selector:
+    name: my-db-selector
+----
+====
+. Create a LoadBalancer service on your pod:
++
+----
+$ oc create -f loadbalancer.yaml
+----
+. Check the service for an external IP. For example, for a service named `myservice`:
++
+----
+$ oc get myservice
+----
++
+When your LoadBalancer-type service has an external IP assigned, the output
+displays the IP:
++
+----
+NAME         CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
+myservice    172.30.74.106   172.46.0.1    80/TCP    30s
+----
++
+However, if your LoadBalancer-type service does not have a properly-assigned
+ingress IP, the output shows it as pending:
```

Comment:
since the default  'ingressIPNetworkCIDR' is 172.46.0.0/16.  so if not setting 'loadBalancerIP' in service yaml file. the 'EXTERNAL-IP' will be assigned one ip automatically from the range.